src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -978,8 +978,12 @@ private Expression AddOrderByQueryForSource(Expression source, OrderByClause ord
 
         // new CollectionWrapper<ElementType> { Instance = source.Select((ElementType element) => new Wrapper { }) }
         [SuppressMessage("Microsoft.Maintainability", "CA1502:AvoidExcessiveComplexity", Justification = "These are simple conversion function and cannot be split up.")]
-        private Expression ProjectCollection(Expression source, Type elementType,
-            SelectExpandClause selectExpandClause, IEdmStructuredType structuredType, IEdmNavigationSource navigationSource,
```

Comment:
How about the "skip" since it's at same level as "top"?

Reply:
Skip should be fixed as well since it's using modifiedSource, although I don't have a test for that.